site/src/request_handlers/graph.rs

```diff
@@ -112,6 +112,13 @@ async fn create_graphs(
             .insert(scenario, graph_series);
     }
 
+    // Keep only benchmarks that have some data available
+    benchmarks.retain(|_, benchmark| {
+        benchmark
+            .values()
+            .any(|profile| profile.values().any(|series| !series.points.is_empty()))
+    });
```

Comment:
I can move it there, but it seems a bit weird to me that such a low level function (`execute_query`) would do some postprocessing that would hide data. The motivation for this PR was not avoiding processing of data on the backend, but simply hiding empty graphs in the UI. It seems more fitting to me to do that in the page/UI specific code rather than in a function shared by multiple pages. Do we actually want to hide this data from all consumers?

Reply:
I changed to the approach that you have suggested. While at it, I took the liberty of slightly refactoring the code to use `zip` instead of manual indexing.